props/saber_BC_buttons.h

```diff
@@ -1567,6 +1589,25 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
     if (scroll_presets_beep_delay_timer_.isTimerExpired()) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
+    // Blade array changing helper. See id() for usage.
+    if (play_no_change_sound_) {
+      play_no_change_sound_ = false;
+
+      float real_id = PropBase::id(false);
+      size_t real_best_config = FindBestConfigForId(real_id);
+
+      if (real_best_config == 0) {
+        // Real blade event resulted in no blade array, so blade removal
+        if (!hybrid_font.PlayPolyphonic(&SFX_bladeout)) {
```

Comment:
I don't understand why we would play bladeout/bladein if there was no change?
Shouldn't this play the array sound?

Reply:
Even if the blade array doesn’t actually change, I'm trying to distinguish between blade insertion and removal.
Yes, this is assumed that NO_BLADE is first in the BladeConfig.
That's how I always run it. I could add a "rule" to the prop comments saying that. But if it's unacceptable then I need to work on it, possibly again unguarding blade_detected_.

Reply:
So, are you trying to capture the difference between the current state and the previous "fake" state?
Or are you thinking something else?

Reply:
Again, I was going with "if a real blade is removed, we will be arriving on NO_BLADE, so must be bladein.wav.
If a blade is inserted, we will be switching to something *not* NO_BLADE, so play bladein.wav."
It is working.
I did change 
```cpp
if (real_best_config == 0) {
        // Real blade event resulted in no blade array, so blade removal
        ```
        
   to use `current_config->ohm >= NO_BLADE` instead, so as not to force NO_BLADE to be the first array.

Reply:
I think I'm a bit slow today, it makes sense now.

Reply:
Oh, well I just found that I still get font.wav if going from an array that's not zero, to another one (like be on array 1 , insert blade, go to array 3, but font.wav plays instead of expected bladein.wav. 
I can't nip it.